modules/accredited_representative_portal/app/sidekiq/accredited_representative_portal/redact_power_of_attorney_requests_job.rb

```diff
@@ -0,0 +1,147 @@
+
+      PowerOfAttorneyRequest
+        .where(id: combined_ids)
+        .includes(:resolution, :power_of_attorney_form, :power_of_attorney_form_submission)
+    end
+
+    def expired_request_ids
+      resolution_table_name = PowerOfAttorneyRequestResolution.table_name
+
+      PowerOfAttorneyRequest
+        .joins(:resolution)
+        .unredacted
+        .where(resolution_table_name => {
+                 resolving_type: 'AccreditedRepresentativePortal::PowerOfAttorneyRequestExpiration'
+               }).ids
+    end
+
+    def stale_processed_request_ids
+      threshold = Time.current - STALE_RESOLUTION_DURATION
+      resolution_table_alias = 'resolution' # The alias defined in processed_join_sql
+
+      PowerOfAttorneyRequest
+        .processed
+        .unredacted
+        .where("#{resolution_table_alias}.created_at < ?", threshold)
+        .where.not(
+          "#{resolution_table_alias}.resolving_type = ?",
+          'AccreditedRepresentativePortal::PowerOfAttorneyRequestExpiration'
+        ).ids
+    end
+
+    def process_requests(requests_to_process)
+      results = { redacted: 0, errors: 0 }
+
+      requests_to_process.each do |request|
+        if attempt_redaction(request)
+          results[:redacted] += 1
+        else
+          results[:errors] += 1
+        end
+      end
+
+      results
+    end
+
+    # Attempts to redact a single request, handling potential errors
+    # Returns true on success, false on failure
+    def attempt_redaction(request)
+      redact_request(request)
+      true
+    rescue => e
+      log_redaction_error(request, e)
+      false
+    end
+
+    # Performs the actual redaction logic for a single request within a transaction
+    def redact_request(request)
+      request.transaction do
+        log_request_redaction(request)
+
+        form = request.power_of_attorney_form
+        submission = request.power_of_attorney_form_submission
+        resolution = request.resolution
+
+        # 1. Destroy associated form
+        form&.destroy!
+
+        # 2. Redact submission data using update_columns direct to the db
+        # we're using #update_columns to skip validation because the redaction
+        # removes required fields and will leave the record invalid
+        # rubocop:disable Rails/SkipsModelValidations
+        if submission.present?
+          submission.update_columns(
+            # Use the actual ciphertext/key column names
+            service_response_ciphertext: nil,
+            error_message_ciphertext: nil,
+            encrypted_kms_key: nil
+          )
+        end
+
+        # 3. Redact resolution data using update_columns
+        if resolution.present?
+          resolution_updates = { reason_ciphertext: nil }
+          # Add the key to the hash only if the column exists on the resolution table
+          resolution_updates[:encrypted_kms_key] = nil if resolution.has_attribute?(:encrypted_kms_key)
```

Comment:
But only one table is being manipulated here right? `ar_power_of_attorney_request_resolutions`?